src/activity/activity.ts

```diff
@@ -244,25 +255,25 @@ export class Activity {
           }
 
           try {
-            logger.debug("Trying to poll for batch execution results from yagna");
-            // This will ignore "incompatibility" between ExeScriptCommandResultResultEnum and ResultState, which both
-            // contain exactly the same entries, however TSC refuses to compile it as it assumes the former is dynamically
-            // computed.
-            const rawExecBachResults = await api.activity.control.getExecBatchResults(
-              activityId,
-              batchId,
-              undefined,
-              activityExeBatchResultPollIntervalSeconds,
+            const rawExecBachResults = await retry(
+              async (bail, attempt) => {
+                logger.debug(`Trying to poll for batch execution results from yagna. Attempt: ${attempt}`);
+                return api.activity.control
+                  .getExecBatchResults(activityId, batchId, undefined, activityExeBatchResultPollIntervalSeconds)
+                  .catch((err) => handleError(err, bail, attempt));
```

Comment:
I don't think that the bail method should ever go out to the `handleError` method. The `handleError` can be used in many other contexts where the bail doesn't make sense. Instead, within the `retry` I would call `bail`, this will lead to a throw of the error and then I would pass the error to the `handleError` function.